frontend/src/v5/ui/components/shared/appBar/appBar.component.tsx

```diff
@@ -15,31 +15,153 @@
+import { AppBar as MuiAppBar, ClickAwayListener, Grow } from '@material-ui/core';
 
 import LogoIcon from '@assets/icons/logo.svg';
 import IntercomIcon from '@assets/icons/intercom.svg';
 import NotificationsIcon from '@assets/icons/notifications.svg';
 import { CircleButton } from '@/v5/ui/controls/circleButton';
-import { AvatarButton } from '@/v5/ui/controls/avatarButton';
-import { Items } from './appBar.styles';
+import ContactUsIcon from '@assets/icons/email.svg';
+import InviteAFriendIcon from '@assets/icons/add_user.svg';
+import TeamspacesIcon from '@assets/icons/teamspaces.svg';
+import VisualSettingsIcon from '@assets/icons/settings.svg';
+import SupportCentreIcon from '@assets/icons/question_mark.svg';
+import { CurrentUserHooksSelectors } from '@/v5/services/selectorsHooks/currentUserSelectors.hooks';
+import { AvatarButton } from '@controls/avatarButton';
+import {
+	Items,
+	UserMenu,
+	Section,
+	AvatarSection,
+	Popper,
+	Paper,
+	UserFullName,
+	UserUserName,
+	SignOutButton,
+	EditProfileButton,
+	Avatar,
+} from './appBar.styles';
 import { Breadcrumbs } from '../breadcrumbs';
+import { UserMenuButton } from './userMenuButton/userMenuButton.component';
 
-export const AppBar = (): JSX.Element => (
-	<MuiAppBar position="static">
-		<Items>
-			<LogoIcon />
-			<Breadcrumbs />
-		</Items>
-		<Items>
-			<CircleButton variant="contrast" aria-label="intercom">
-				<IntercomIcon />
-			</CircleButton>
-			<CircleButton variant="contrast" aria-label="notifications">
-				<NotificationsIcon />
-			</CircleButton>
-			<AvatarButton> GH </AvatarButton>
-		</Items>
-	</MuiAppBar>
+const getUserNamesInitials = ({ firstName, lastName }) => (
+	[firstName, lastName]
+		.map((name) => name.trim().charAt(0).toUpperCase())
+		.join('')
 );
+
+export const AppBar = (): JSX.Element => {
+	const { teamspace } = useParams();
+	const { url } = useRouteMatch();
+	const baseUrl = url.split('/').slice(0, 3).join('/');
+	const userIsPending = CurrentUserHooksSelectors.selectIsPending();
+	const user = CurrentUserHooksSelectors.selectCurrentUser();
+
+	const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+	const handleCloseDropdown = () => {
+		setAnchorEl(null);
+	};
+
+	const handleClickDropdown = (event: MouseEvent<HTMLButtonElement>) => {
+		event.stopPropagation();
+		setAnchorEl(event.currentTarget);
+	};
+
+	return (
+		<>
+			<MuiAppBar position="static">
+				<Items>
+					<LogoIcon />
+					<Breadcrumbs />
+				</Items>
+				<Items>
+					<CircleButton variant="contrast" aria-label="intercom">
+						<IntercomIcon />
+					</CircleButton>
+					<CircleButton variant="contrast" aria-label="notifications">
+						<NotificationsIcon />
+					</CircleButton>
+					<AvatarButton
+						onClick={handleClickDropdown}
+						src={user.hasAvatar ? user.avatarUrl : null}
+					>
+						{!userIsPending && getUserNamesInitials(user)}
+					</AvatarButton>
+				</Items>
+			</MuiAppBar>
+			<Popper
```

Comment:
Done. The new component is called `ActionMenu` and it accepts a `TriggerButton` as well as other children.
```js
<ActionMenu>
     <TriggerButton>
          <Avatar />  <-- must be wrapped in the TriggerButton parent and does not need any props
     </TriggerButton>
     <ActionMenuSection>
         <ActionMenuItem>  <-- can be nested
            ...
         </ActionMenuItem>
     </ActionMenuSection>
     ....
</ActionMenuItem>
```
 The `ActionMenu` automatically checks that the `TriggerButton` is passed and then adds the onclick functionality to open the menu. 

The `ActionMenu` also inspects all the elements being passed to it searching for all the `ActionMenuButton`s and `ActionMenuItem`s. These components' onclick function is then extended so that, after they are clicked, the menu gets closed. 

The `ActionMenu` component has a bit of "fancy" logic, but it is completely autonomous (does not need `onClickClose`, `anchorEl` or similar props as it handles all these aspects itself.